### PR TITLE
Email extraction omitted for broken contact links

### PR DESCRIPTION
**Pull request by In-vincible**
This solves #2187.
This PR, skips the email data for broken links, and lets the scraper continue.
I have created #2189, for the broken link, if we want to get the info in any case.

### CODE REVIEW

File: openstates/mi/people.py [hunk 1]
```diff
@@ -15,6 +16,7 @@ def scrape(self, chamber=None, session=None):
         elif chamber == 'lower':
             yield from self.scrape_lower(chamber)
         else:
+            requests.exceptions.TooManyRedirects
```
Comment (mileswwatkins) [inline openstates/mi/people.py]: What does this statement do?

File: openstates/mi/people.py [hunk 2]
```diff
@@ -136,25 +138,27 @@ def scrape_upper(self, chamber):
...
+                else:
+                    # not using the most common template, but maybe they
+                    # dropped their email on the page somewhere
+                    links = contact_doc.xpath('//a') or []
+                    text_email = [a for a in links
+                                  if 'mailto:' in (a.get('href') or '')]
+                    if text_email:
+                        email = text_email[0].text
+            except requests.exceptions.TooManyRedirects:
+                print("Conatct Link Not Working for %s" % name)
```
Comment (mileswwatkins) [inline openstates/mi/people.py]: Instead of `print("message")`, can you use `self.warning("message")`? This uses more formalized Python `logging`.

Comment (mileswwatkins) [reply openstates/mi/people.py]: Also, small typo in the error message itself.

**Comment by In-vincible** [comment]
@mileswwatkins Really Sorry I think this commit, was rather carelessly done, was traveling, I will put forth a PR with the changes.

**Comment by mileswwatkins** [comment]
Oh man, I made a mistake too, must've hit "Merge" before I meant to, haha. Thanks for the follow-up PR!

**Pull request merged**